hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonOutputCommitter.java

```diff
@@ -302,6 +317,54 @@ private void commitJobForPartition(JobContext context, boolean overwriteSet,
     commitJobFinal(context, loadModel, operationContext, carbonTable, uniqueId);
   }
 
+  @SuppressWarnings("unchecked")
+  private void writeSegmentWithoutMergeIndex(JobContext context, CarbonLoadModel loadModel,
+      String segmentFileName, String partitionPath) throws IOException {
+    Map<String, String> IndexFileNameMap = (Map<String, String>) ObjectSerializationUtil
+        .convertStringToObject(context.getConfiguration().get("carbon.index.files.name"));
+    List<String> partitionList =
+        (List<String>) ObjectSerializationUtil.convertStringToObject(partitionPath);
+    SegmentFileStore.SegmentFile finalSegmentFile = null;
+    boolean isRelativePath;
+    String path;
+    for (String partition : partitionList) {
+      isRelativePath = false;
+      path = partition;
+      if (path.startsWith(loadModel.getTablePath())) {
+        path = path.substring(loadModel.getTablePath().length());
+        isRelativePath = true;
+      }
+      SegmentFileStore.SegmentFile segmentFile = new SegmentFileStore.SegmentFile();
+      SegmentFileStore.FolderDetails folderDetails = new SegmentFileStore.FolderDetails();
+      Set<String> set = new HashSet<String>();
+      set.add(IndexFileNameMap.get(partition));
+      folderDetails.setFiles(set);
+      List<String> partitions = new ArrayList<String>();
```

Comment:
```suggestion
      List<String> partitions = Collections.singletonList(path.substring(path.indexOf("/") + 1));```
and remove next line

Reply:
Modified